### ISSUE TITLE
Support QUERY method

### ISSUE THREAD

**Issue by jeffparsons** [hyperium/hyper#2748]
The [QUERY method](https://www.ietf.org/id/draft-ietf-httpbis-safe-method-w-body-02.html) is not yet standardized, but it looks likely that it will be.

Is the Hyper project open to supporting not-yet-standard HTTP features, perhaps behind a feature flag? Or would you prefer to wait until it is actually standardized?

**Comment by seanmonstar** [comment]
Is there anything special that hyper needs to do? I'm vaguely familiar with the proposal. It should just work, no?

**Comment by jeffparsons** [comment]
I don't believe Hyper needs to do anything other than treat QUERY exactly the same as GET, assuming it currently accepts bodies on GET requests.

But I must stress that I've only read secondary sources on this — i.e. I've yet to read the draft spec itself.

**Comment by seanmonstar** [comment]
To my knowledge, it should work fine in hyper already. :)
